db/migrate/20180111162442_move_pictures_blobs_to_pictures.rb

```diff
@@ -18,7 +18,7 @@ def up
 
     say_with_time("Moving picture content from BinaryBlobs to the pictures table") do
       BinaryBlob.in_my_region.includes(:binary_blob_parts).where(:resource_type => "Picture").find_each do |blob|
-        Picture.update(blob.resource_id, :content => blob.data, :extension => blob.data_type, :md5 => blob.md5) if blob.resource_id
+        Picture.find_by(:id => blob.resource_id).try(:update_attributes!, :content => blob.data, :extension => blob.data_type, :md5 => blob.md5)
```

Comment:
I think it would be better to do

```ruby
Picture.where(:id => blob.resource_id).update_all(:content => blob.data, :extension => blob.data_type, :md5 => blob.md5)
```

so you don't have to pull the Picture object and then update (2 queries + in memory object - basically an N+1)

Reply:
👍 Yep, that's what I was trying to avoid all along.  For some reason I didn't think to `update_all` with a `where`.